doc/message_dispatching.puml

```diff
@@ -0,0 +1,52 @@
+
+@startuml
+
+interface MessageDispatcherInterface {
+    +dispatch_call(const json& call, bool triggered = false)
+    +dispatch_call_async(const json& call, bool triggered = false): std::future<EnhancedMessage<T>>
+    +dispatch_call_result(const json& call_result)
+    +dispatch_call_error(const json& call_error)
+    +new_message_id(): MessageId
+}
+
+class v16::MessageDispatcher {
+    - MessageQueue& message_queue
+    - ChargePointConfiguration& configuration
+    - RegistrationStatus& registration_status
+}
+
+class v201::MessageDispatcher {
+    - MessageQueue& message_queue
+    - DeviceModel& device_model
+    - ConnectivityManager& connectivity_manager
+    - RegistrationStatusEnum& registration_status
+}
+
+interface v201::DataTransferInterface {
+    +data_transfer_req(request: DataTransferRequest): std::optional<DataTransferResponse>
+    +handle_data_transfer_req(call: Call<DataTransferRequest>)
```

Comment:
Should we replace this with a `handle_message` function that takes an `EnhancedMessage<v201::MessageType>`?

For this class there is only one message to handle but ideally that should not change the API for these functional blocks. If we later have to add a message to handle we can do it purely in the functional block